bundles/org.openhab.binding.bluetooth/src/main/java/org/openhab/binding/bluetooth/discovery/internal/BluetoothDiscoveryService.java

```diff
@@ -296,6 +303,9 @@ private void publishDiscoveryResult(BluetoothAdapter adapter, DiscoveryResult re
 
         private void retractDiscoveryResult(BluetoothAdapter adapter, DiscoveryResult result) {
             Set<DiscoveryResult> results = discoveryResults.remove(adapter);
+            // The method `thingRemoved()` removes the Thing from listeners like `Inbox`.
+            // We therefore need to reset `latestSnapshot` so that the Thing is notified again next time.
+            latestSnapshot.putValue(null);
```

Comment:
I don't think you should be removing the snapshot here. `retractDiscoveryResult` is only called when a new discovery would be published and thus required the old discovery to be removed first. By clearing the snapshot here you are forcing the discovery process to be rerun unnecessarily.

I think clearing the caches as part of stopScan should be more than sufficient for this PR.

Reply:
I see.. Sound reasonable, I added your explanation as comment and removed the code.